src/entity.cpp

```diff
@@ -656,213 +656,245 @@ void Entity_CheckCollisionCallbacks(entity_p ent)
 
 void Entity_DoAnimCommands(entity_p entity, struct ss_animation_s *ss_anim, int changing)
 {
-    if((World_GetAnimCommands() == NULL) || (ss_anim->model == NULL))
+    if(World_GetAnimCommands() && ss_anim->model && (changing >= 0x01))
     {
-        return;  // If no anim commands
-    }
-
-    animation_frame_p af  = ss_anim->model->animations + ss_anim->current_animation;
-    if(af->num_anim_commands <= 255)
-    {
-        uint32_t count        = af->num_anim_commands;
-        int16_t *pointer      = World_GetAnimCommands() + af->anim_command;
-        int8_t   random_value = 0;
-
-        for(uint32_t i = 0; i < count; i++, pointer++)
+        animation_frame_p next_af = ss_anim->model->animations + ss_anim->next_animation;
+        if(next_af->num_anim_commands <= 255)
         {
-            switch(*pointer)
-            {
-                case TR_ANIMCOMMAND_SETPOSITION:
-                    // This command executes ONLY at the end of animation.
-                    pointer += 3; // Parse through 3 operands.
-                    break;
+            float params[3];
+            uint32_t count        = next_af->num_anim_commands;
+            int16_t *pointer      = World_GetAnimCommands() + next_af->anim_command;
+            int8_t   random_value = 0;
 
-                case TR_ANIMCOMMAND_JUMPDISTANCE:
-                    // This command executes ONLY at the end of animation.
-                    pointer += 2; // Parse through 2 operands.
-                    break;
-
-                case TR_ANIMCOMMAND_EMPTYHANDS:
-                    ///@FIXME: Behaviour is yet to be discovered.
-                    break;
-
-                case TR_ANIMCOMMAND_KILL:
-                    // This command executes ONLY at the end of animation.
-                    if(ss_anim->current_frame == af->frames_count - 1)
-                    {
-                        if(entity->character)
+            for(uint32_t i = 0; i < count; i++, pointer++)
```

Comment:
to parse all commands in all available places I will add some hack to the cycle (I hope I will found solution without hack)